src/MoBi.Core/Domain/Services/SimulationFactory.cs

```diff
@@ -64,5 +84,62 @@ public IMoBiSimulation Create()
       {
          return CreateFrom(_simulationConfigurationFactory.Create(), null);
       }
+
+      private void validateDimensions(IModel model, SimulationBuilder simulationBuilder)
+      {
+         _dimensionValidator.Validate(model, simulationBuilder)
+            .SecureContinueWith(t => showWarnings(t.Result));
+      }
+
+      public IModel CreateModelAndValidate(string modelName, SimulationConfiguration simulationConfiguration)
+      {
+         var results = createModel(simulationConfiguration, modelName);
+
+         if (results == null || results.IsInvalid)
+            throw new MoBiException(AppConstants.Exceptions.CouldNotCreateSimulation);
+
+         validateDimensions(results.Model, results.SimulationBuilder);
+
+         return results.Model;
+      }
+
+      public IMoBiSimulation CreateSimulationAndValidate(SimulationConfiguration configuration, string name)
```

Comment:
wondering why CreateSImulationAndValidate does not use CreateModelAndValidate. Almost the same aside from the heavyWorkManager